File: backend/app/schemas/stock.py
from datetime import datetime

from pydantic import BaseModel


class StockFollowRequest(BaseModel):
    ticker: str


class StockSummaryResponse(BaseModel):
    id: int
    ticker: str
    company_name: str
    exchange: str
    sector: str | None
    industry: str | None
    current_price_inr: float | None
    market_cap_inr: float | None
    pe_ratio: float | None
    dividend_yield: float | None
    debt_to_equity: float | None
    fundamentals_summary: str | None
    rolling_sentiment_label: str | None = None
    rolling_sentiment_score: float | None = None


class FollowedStockResponse(BaseModel):
    id: int
    is_active: bool
    followed_at: datetime
    last_ingested_at: datetime | None
    stock: StockSummaryResponse


class StockIngestionResponse(BaseModel):
    followed_stock: FollowedStockResponse
    ingested_documents: int
    ingested_chunks: int
    message: str


class StockRefreshResponse(StockIngestionResponse):
    pass

File: frontend/src/types/stock.ts
export interface StockSummary {
  id: number
  ticker: string
  company_name: string
  exchange: string
  sector: string | null
  industry: string | null
  current_price_inr: number | null
  market_cap_inr: number | null
  pe_ratio: number | null
  dividend_yield: number | null
  debt_to_equity: number | null
  fundamentals_summary: string | null
  rolling_sentiment_label: string | null
  rolling_sentiment_score: number | null
}

export interface FollowedStock {
  id: number
  is_active: boolean
  followed_at: string
  last_ingested_at: string | null
  stock: StockSummary
}

export interface StockIngestionResponse {
  followed_stock: FollowedStock
  ingested_documents: number
  ingested_chunks: number
  message: string
}

export interface StockUnfollowResponse {
  ticker: string
  message: string
}

export interface StockDetailDocument {
  ticker: string
  title: string
  source_type: string
  url: string | null
  published_at: string | null
  snippet: string
  sentiment_label: string | null
  sentiment_score: number | null
  impact_label: string | null
  event_type: string | null
  similarity_score: number | null
}

export interface StockDetailResponse {
  stock: StockSummary
  is_followed: boolean
  followed_at: string | null
  last_ingested_at: string | null
  documents: StockDetailDocument[]
}
